src/state/NetworkManager.ts

```diff
@@ -108,23 +114,18 @@ export default class NetworkManager implements Manager {
    * Will resolve the promise associated with receive url.
    */
   protected handleReceive(action: ReceiveAction) {
-    const completePromise = () => {
-      // this can still turn out to be untrue since this is async
-      if (action.meta.url in this.fetched) {
-        let promiseHandler: (value?: any) => void;
-        if (action.error) {
-          promiseHandler = this.rejectors[action.meta.url];
-        } else {
-          promiseHandler = this.resolvers[action.meta.url];
-        }
-        promiseHandler(action.payload);
-        // since we're resolved we no longer need to keep track of this promise
-        this.clear(action.meta.url);
+    // this can still turn out to be untrue since this is async
+    if (action.meta.url in this.fetched) {
+      let promiseHandler: (value?: any) => void;
+      if (action.error) {
+        promiseHandler = this.rejectors[action.meta.url];
+      } else {
+        promiseHandler = this.resolvers[action.meta.url];
       }
-    };
-    // TODO: this should call after the reducer has been updated
-    // in all concurrent fibers
-    RIC(completePromise, { timeout: 1000 });
+      promiseHandler(action.payload);
```

Comment:
this code is not only prettier...it's now guaranteed bug-free in concurrent mode.